impl/src/main/java/org/ehcache/config/builders/CacheManagerBuilder.java

```diff
@@ -68,37 +70,46 @@ public T build() {
     return build(false);
   }
 
-  private CacheManagerBuilder() {
+  protected CacheManagerBuilder() {
     this.configBuilder = newConfigurationBuilder();
     this.services = emptySet();
   }
-
-  private CacheManagerBuilder(CacheManagerBuilder<T> builder, Set<Service> services) {
+  
+  protected CacheManagerBuilder(CacheManagerBuilder<T> builder, Set<Service> services) {
     this.configBuilder = builder.configBuilder;
     this.services = unmodifiableSet(services);
   }
 
-  private CacheManagerBuilder(CacheManagerBuilder<T> builder, ConfigurationBuilder configBuilder) {
+  protected CacheManagerBuilder(CacheManagerBuilder<T> builder, ConfigurationBuilder configBuilder) {
     this.configBuilder = configBuilder;
     this.services = builder.services;
   }
 
-  public static CacheManager newCacheManager(final Configuration configuration) {
-    return new EhcacheManager(configuration);
+  protected CacheManagerBuilder(final CacheManagerBuilder<?> builder) {
+    this.configBuilder = builder.configBuilder;
+    this.services = builder.services;
   }
 
-  T newCacheManager(Collection<Service> services, final Configuration configuration) {
+  protected T newCacheManager(Collection<Service> services, final Configuration configuration) {
     final EhcacheManager ehcacheManager = new EhcacheManager(configuration, services);
     return cast(ehcacheManager);
   }
 
+  protected CacheManagerBuilder<T> newBuilder(final Set<Service> services) {
+    return new CacheManagerBuilder<T>(this, services);
+  }
+
+  protected CacheManagerBuilder<T> newBuilder(final ConfigurationBuilder builder) {
+    return new CacheManagerBuilder<T>(this, builder);
+  }
+
   @SuppressWarnings("unchecked")
-  T cast(EhcacheManager ehcacheManager) {
+  private T cast(EhcacheManager ehcacheManager) {
     return (T) ehcacheManager;
   }
 
   public <K, V> CacheManagerBuilder<T> withCache(String alias, CacheConfiguration<K, V> configuration) {
-    return new CacheManagerBuilder<T>(this, configBuilder.addCache(alias, configuration));
+    return this.newBuilder(configBuilder.addCache(alias, configuration));
```

Comment:
For of habit from many years spent programming Java where `this` was a requirement ...